Lib/slapdtest/_slapdtest.py

```diff
@@ -596,3 +623,43 @@ def setUpClass(cls):
     @classmethod
     def tearDownClass(cls):
         cls.server.stop()
+
+
+class PPolicyEnabledSlapdObject(SlapdObject):
+    """
+    A subclass of :py:class:`SlapdObject` with password policy enabled.
+    Note that this class has no actual password policy configuration entries.
+    It is the job of the users of this class to define
+    the default password policies on their own.
+    The dn of the default is :attr:`.default_ppolicy_dn` of this class.
+    """
+
+    openldap_schema_files = (
+        'core.schema', 'ppolicy.schema'
+    )
+    modules = (
+        'ppolicy',
+    )
+
+    default_ppolicy_dn = "cn=default-ppolicy,%(suffix)s" % {
+        'suffix': SlapdObject.suffix
+    }
+
+    overlays = (
+        {
+            'name': 'ppolicy',
+            'configuration': "\n".join([
+                'ppolicy_default "{}"'.format(default_ppolicy_dn),
+                # let slapd tell the clients that they are locked out
+                'ppolicy_use_lockout'])
+        },
+    )
+
+
+class PPolicyEnabledSlapdTestCase(SlapdTestCase):
+    """
+    A subclass of :py:class:`SlapdTestCase`, which uses
+    :py:class:`PPolicyEnabledSlapdObject` as the slapd controller.
+    """
+
+    server_class = PPolicyEnabledSlapdObject
```

Comment:
Is this necessary? `PPolicyEnabledSlapdTestCase` needs to be subclassed; the subclass can set its `server_class` directly.